homeassistant/components/sma/sensor.py

```diff
@@ -226,4 +243,14 @@ def async_update_values(self):
     @property
     def unique_id(self):
         """Return a unique identifier for this sensor."""
-        return f"sma-{self._sensor.key}-{self._sensor.name}"
+        return f"sma-{self._confg_entry_unique_id}-{self._sensor.key}"
```

Comment:
This old key was not unique with multiple inverters, but this seems ok now?

Reply:
Yes this is unique. Tested this by adding multiple devices and it will create two sensors. One will have a _2 appended automatically,

Reply:
I can help test if needed, I ended up with a custom solution but should be easy to use the HA SMA integration as well.